ec/ecresource/deploymentresource/schema_enteprise_search.go

```diff
@@ -107,3 +111,44 @@ func enterpriseSearchTopologySchema() *schema.Schema {
 		},
 	}
 }
+
+func enterpriseSearchConfig() *schema.Schema {
+	return &schema.Schema{
+		Type:             schema.TypeList,
+		Optional:         true,
+		MaxItems:         1,
+		DiffSuppressFunc: suppressMissingOptionalConfigurationBlock,
+		Description:      `Optionally define the Enterprise Search configuration options for the Enterprise Search Server`,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"secret_session_key": {
+					Type:        schema.TypeString,
+					Description: `Optionally override the secret session key within Enterprise Search - defaults to the previously existing secretSession`,
+					Computed:    true,
+				},
+
+				// User settings
```

Comment:
I don't think this line is necessary? the fields already start with `"user_settings_*"`